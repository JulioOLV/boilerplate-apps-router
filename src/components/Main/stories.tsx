import { Meta, StoryObj } from '@storybook/nextjs'
import Main from '.'

const meta = {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta<typeof Main>
export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {}

export const Basic: Story = {
  args: {
    title: 'title basic',
    description: 'description basic'
  }
}
