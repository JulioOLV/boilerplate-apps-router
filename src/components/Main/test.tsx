import { screen, render } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('render correctly', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
