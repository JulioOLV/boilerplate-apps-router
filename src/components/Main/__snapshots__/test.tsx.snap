// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> render correctly 1`] = `
<main
  class="sc-bcPKhP jhrJfq"
>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
